app/Http/Controllers/API/TraderController.php

```diff
@@ -173,6 +179,20 @@ public function emailVoucherHistory(Request $request, Trader $trader)
         return response()->json(['message' => $response_text], 202);
     }
 
+    public function getMinMaxVoucherDates($vouchers) {
```

Comment:
I think this should be in the Voucher class not the trader class. Mix Mx dates should be determined for a group of vouchers and has nothing to do with Trader.